pkg/controller/instance/instance_controller.go

```diff
@@ -350,12 +367,27 @@ func createPlan(mgr manager.Manager, planName string, instance *kudov1alpha1.Ins
 		return err
 	}
 
-	if err := mgr.GetClient().Create(context.TODO(), &planExecution); err != nil {
+	if err := mgr.GetClient().Create(ctx, &planExecution); err != nil {
 		log.Printf("InstanceController: Error creating planexecution \"%v\": %v", planExecution.Name, err)
 		recorder.Event(instance, "Warning", "CreatePlanExecution", fmt.Sprintf("Error creating planexecution \"%v\": %v", planExecution.Name, err))
 		return err
 	}
 	recorder.Event(instance, "Normal", "PlanCreated", fmt.Sprintf("PlanExecution \"%v\" created", planExecution.Name))
+
+	//update the instance with plan
+	active := &instance.Status.ActivePlan
+	log.Printf("PlanExecutionController: Updating active plan of %v", active)
+	active.Name = planExecution.Name
+	active.Kind = planExecution.Kind
+	active.Namespace = planExecution.Namespace
+	active.APIVersion = planExecution.APIVersion
+	active.UID = planExecution.UID
+	log.Printf("PlanExecutionController: Updating active plan to %v", instance.Status.ActivePlan)
+	if err := mgr.GetClient().Update(ctx, instance); err != nil {
+		log.Printf("InstanceController: Error updating Instance \"%v\": %v", instance.Name, err)
+		return err
+	}
+
```

Comment:
Would be great making this a recorded event
```suggestion
recorder.Event(instance, "Normal", "PlanUpdated", fmt.Sprintf("Updating active plan %s to %s", active, instance.Status.ActivePlan))
```